test/server/runner-test.js

```diff
@@ -325,4 +437,39 @@ describe('Runner', function () {
             testFilter(filter, expectedTestNames, done);
         });
     });
+
+    describe('.run()', function () {
+        it('Should raise exception if connection disconnected during tests', function (done) {
+            var brokenConnection = testCafe.createBrowserConnection();
+
+            brokenConnection.establish('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 ' +
+                                       '(KHTML, like Gecko) Chrome/41.0.2227.1 Safari/537.36');
+
+            var run = runner
+                .browsers(brokenConnection)
+                .reporter('list')
+                .src('test/server/data/test-suite/top.test.js')
+                .run()
+                .then(function () {
+                    throw new Error('Promise rejection expected');
+                })
+                .catch(function (err) {
+                    expect(err.message).eql('I have failed :(');
+                });
+
+            setTimeout(function () {
+                var status = brokenConnection.getStatus();
+
+                expect(status.cmd).eql(COMMAND.run);
```

Comment:
We shouldn't assert status here, we need to repeat timeout until we will not get `COMMAND.run`, then emit error